src/mlpack/bindings/python/print_input_processing.hpp

```diff
@@ -108,35 +108,82 @@ void PrintInputProcessing(
 {
   const std::string prefix(indent, ' ');
 
+  std::cout << prefix << "# Detect if the parameter was passed; set if so."
+      << std::endl;
+  if (!d.required)
+  {
+    if (T::is_row || T::is_col)
+    {
   /**
    * This gives us code like:
    *
    * # Detect if the parameter was passed; set if so.
    * if param_name is not None:
    *   param_name_tuple = to_matrix(param_name)
-   *   param_name_mat = arma_numpy.numpy_to_mat_d(param_name_tuple[0],
+   *   if param_name_tuple[0].shape[0] == 1 or
+   *       param_name_tuple[0].shape[1] == 1:
+   *     param_name_tuple[0] = param_name_tuple[0].ravel()
+   *   param_name_mat = arma_numpy.numpy_to_rol_s(param_name_tuple[0],
+   *	   param_name_tuple[1])
+   *   SetParam[mat](<const string> 'param_name', dereference(param_name_mat))
+   *   CLI.SetPassed(<const string> 'param_name')
+   */
+      std::cout << prefix << "if " << d.name << " is not None:" << std::endl;
+      std::cout << prefix << "  " << d.name << "_tuple = to_matrix(" << d.name
+          << ", dtype=" << GetNumpyType<typename T::elem_type>() << ", "
+          << "copy=CLI.HasParam('copy_all_inputs'))" << std::endl;
+      std::cout << prefix << "  " << "if len(" << d.name << "_tuple[0].shape"
+          << ") > 1:" << std::endl;
+      std::cout << prefix << "  " << prefix << "if " << d.name << "_tuple[0]"
+          << ".shape[0] == 1 or " << d.name << "_tuple[0].shape[1] == 1:"
+          << std::endl;
+      std::cout << prefix << "  " << prefix << "  " << d.name
+          << "_tuple[0] = " << d.name << "_tuple[0].ravel()" << std::endl;
+      std::cout << prefix << "  " << d.name << "_mat = arma_numpy.numpy_to_"
+          << GetArmaType<T>() << "_" << GetNumpyTypeChar<T>() << "(" << d.name
+          << "_tuple[0], " << d.name << "_tuple[1])" << std::endl;
+      std::cout << prefix << "  SetParam[" << GetCythonType<T>(d)
+          << "](<const string> '" << d.name << "', dereference("
+          << d.name << "_mat))"<< std::endl;
+      std::cout << prefix << "  CLI.SetPassed(<const string> '" << d.name
+          << "')" << std::endl;
+      std::cout << prefix << "  del " << d.name << "_mat";
+    }
+    else
+    {
+  /**
+   * This gives us code like:
+   *
+   * # Detect if the parameter was passed; set if so.
+   * if param_name is not None:
+   *   param_name_tuple = to_matrix(param_name)
+   *   if len(param_name_tuple[0].shape) < 2:
+   *     param_name_tuple[0] = np.reshape((param_name_tuple[0]),
+   *	     (param_name_tuple[0].shape[0] , -1))
```

Comment:
I think you could use `1` instead of `-1` here; there's no need to infer the second dimension when we already know it. :+1: